pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -139,6 +142,11 @@ func NewClusterBuilder(proxy *model.Proxy, req *model.PushRequest, cache model.X
 			cb.credentialSocketExist = true
 		}
 	}
+	if features.MetadataExchange && features.MetadataExchangeAdditionalLabels != "" {
```

Comment:
Sidenote: can we just remove the MetadataExchange feature flag? A bunch of telemetry just breaks without it enabled

Reply:
I think we should discuss this seperately.

Reply:
tracked in https://github.com/istio/istio/issues/54608